### ISSUE TITLE
javancss: Field modifier can not precede annotation

### ISSUE THREAD

**Issue by olivergondza** [cobertura/cobertura#298]
Following declaration produces attached exception using `cobertura-maven-plugin` version 2.7:

```java
private @Deprecated transient String fieldName;
```

```
[ERROR] net.sourceforge.cobertura.javancss.parser.ParseException: Encountered " "transient" "transient "" at line 30, column 25.
Was expecting one of:
    "boolean" ...
    "byte" ...
    "char" ...
    "double" ...
    "float" ...
    "int" ...
    "long" ...
    "short" ...
    "@" ...
    "@" ...
    "@" ...
    "@" ...
    "boolean" ...
    "char" ...
    "byte" ...
    "short" ...
    "int" ...
    "long" ...
    "float" ...
    "double" ...
    "boolean" ...
    "char" ...
    "byte" ...
    "short" ...
    "int" ...
    "long" ...
    "float" ...
    "double" ...
    "boolean" ...
    "char" ...
    "byte" ...
    "short" ...
    "int" ...
    "long" ...
    "float" ...
    "double" ...
    
	at net.sourceforge.cobertura.javancss.parser.JavaParser.generateParseException(JavaParser.java:10490)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.jj_consume_token(JavaParser.java:10366)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.Type(JavaParser.java:2509)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.FieldDeclaration(JavaParser.java:1648)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.ClassBodyDeclaration(JavaParser.java:1071)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.ClassBody(JavaParser.java:898)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.UnmodifiedClassDeclaration(JavaParser.java:824)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.ClassDeclaration(JavaParser.java:728)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.TypeDeclaration(JavaParser.java:575)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.CompilationUnit(JavaParser.java:318)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.parse(JavaParser.java:100)
	at net.sourceforge.cobertura.javancss.Javancss._measureSource(Javancss.java:192)
	at net.sourceforge.cobertura.javancss.Javancss._measureRoot(Javancss.java:268)
	at net.sourceforge.cobertura.javancss.Javancss.<init>(Javancss.java:347)
	at net.sourceforge.cobertura.javancss.JavancssFactory.newInstance(JavancssFactory.java:40)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getFunctionMetricsForSingleFile(ComplexityCalculator.java:243)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getAccumlatedCCNForSingleFile(ComplexityCalculator.java:143)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getCCNForSourceFileNameInternal(ComplexityCalculator.java:273)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getCCNForPackageInternal(ComplexityCalculator.java:208)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getCCNForProject(ComplexityCalculator.java:174)
	at net.sourceforge.cobertura.reporting.xml.XMLReport.<init>(XMLReport.java:70)
	at net.sourceforge.cobertura.reporting.xml.XMLReportFormatStrategy.save(XMLReportFormatStrategy.java:18)
	at net.sourceforge.cobertura.reporting.NativeReport.export(NativeReport.java:31)
	at net.sourceforge.cobertura.reporting.CompositeReport.export(CompositeReport.java:19)
	at net.sourceforge.cobertura.reporting.ReportMain.parseArgumentsAndReport(ReportMain.java:91)
	at net.sourceforge.cobertura.reporting.ReportMain.generateReport(ReportMain.java:141)
	at net.sourceforge.cobertura.reporting.ReportMain.main(ReportMain.java:151)
```

Switching the order of `@Deprecated` and `transient` works fine.

**Comment by christ66** [comment]
@olivergondza Thank you for reporting this! I will write a unit test shortly.

**Comment by christ66** [comment]
@olivergondza Thank you again for this bug report. Issue has been fixed in ae8a9053711709572f9a21328b3bcc806440278b.
